local/gugcat/classes/grade_aggregation.php

```diff
@@ -136,4 +138,21 @@ public static function get_rows($course, $modules, $students){
         }
         return $rows;
     }
+
+    public static function require_resit($studentno){
+        global $aggradeid;
+
+        $grade_ = new grade_grade(array('userid' => $studentno, 'itemid' => $aggradeid), true);
+        $grade_->usermodified = $studentno;
```

Comment:
The usermodified is the the userid of the person who modified this grade, if so the current user.